### PR TITLE
Not WIP but far from complete start for TSP API

### PR DESCRIPTION
**Pull request by ntwyman**
## Description

Adding TSP shipment accept/reject handlers in line with ADR0013.

Also moved the http.Handler construction for the internal and public APIs into the pkg/handlers package.

## Reviewer Notes

swagger.yaml needs only a brief sanity check as it is not expected to be final and will be reviewed in more detail in conjunction with the TSP developers. For this PR, what I am most interested in is high level Feedback about the API patterns and the location of the handler configuration

## Verification Steps

* [x] All tests pass.
* [n/a] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely) have been satisfied.
* [n/a] There are no [aXe](https://www.deque.com/products/aXe/) warnings for UI.
* [n/a] This works in IE.

### CODE REVIEW

File: pkg/handlers/tsps.go [hunk 1]
```diff
@@ -21,3 +21,19 @@ type TSPShipmentsHandler HandlerContext
 func (h TSPShipmentsHandler) Handle(params apioperations.TspShipmentsParams) middleware.Responder {
 	return middleware.NotImplemented("operation .IndexTSPs has not yet been implemented")
 }
+
+// TSPAcceptShipmentsHandler accepts a shipment offered to a TSP
+type TSPAcceptShipmentsHandler HandlerContext
+
+// Handle simply returns a NotImplementedError
+func (h TSPAcceptShipmentsHandler) Handle(params apioperations.AcceptShipmentParams) middleware.Responder {
+	return middleware.NotImplemented("operation .IndexTSPs has not yet been implemented")
```
Comment (stangah) [inline pkg/handlers/tsps.go]: Can you edit these messages (and the one above, which is also wrong) to say the right operation name?

File: swagger/api.yaml [hunk 2]
```diff
@@ -58,8 +60,67 @@ paths:
...
+      responses:
+        200:
+          description: returns updated (accepted) shipment object
+          schema:
+            $ref: '#/definitions/Shipment'
+        400:
+          description: invalid request
+        401:
+          description: must be authenticated to use this endpoint
+        403:
```
Comment (stangah) [inline swagger/api.yaml]: I'm unclear, do we need to list `500` if we might return internal service error (which we always might)

File: swagger/api.yaml [hunk 3]
```diff
@@ -58,8 +60,67 @@ paths:
...
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/AcceptShipmentPayload'
+      responses:
+        200:
+          description: returns updated (accepted) shipment object
+          schema:
+            $ref: '#/definitions/Shipment'
+        400:
```
Comment (stangah) [inline swagger/api.yaml]: If `shipmentUUID` doesn't map to an actual shipment, did you want to return `404` or was `400` going to cover it?

File: swagger/api.yaml [hunk 4]
```diff
@@ -229,37 +269,72 @@ definitions:
         enum: # Add other values
           - REGION_6
         example: REGION_6
-  PreMoveSurvey:
+#  PreMoveSurvey:
```
Comment (stangah) [inline swagger/api.yaml]: Does this comment block need to be here?

File: swagger/api.yaml (hunk 2 above)
Comment (ntwyman) [reply swagger/api.yaml]: Only if we're going to explicitly throw it.

File: pkg/handlers/tsps.go (hunk 1 above)
Comment (ntwyman) [reply pkg/handlers/tsps.go]: done

File: swagger/api.yaml (hunk 4 above)
Comment (ntwyman) [reply swagger/api.yaml]: For now - part of the WIPish bit of it.

**Pull request merged**